tests/unit/backend/data/test_LocalDataService.py

```diff
@@ -641,46 +667,41 @@ def test__generateStateId_cache():
     )
 
     localDataService._readPVFile = mock.Mock()
-    fileMock = mock.Mock()
-    localDataService._readPVFile.return_value = fileMock
-    fileMock.get.side_effect = [
-        [0.1],
-        [0.1],
-        [0.1],
-        [0.1],
-        [1],
-        [0.1],
-        [0.1],  # => "9618b936a4419a6e"
-        [0.2],
-        [0.2],
-        [0.2],
-        [0.2],
-        [1],
-        [0.2],
-        [0.2],
-    ]
-    stateSHA1 = "9618b936a4419a6e"
+
+    # Mock the readDetectorState method to return different DetectorState objects
+    def mock_readDetectorState(runId):
+        if runId == "12345":
+            return DetectorState(arc=(0.1, 0.1), wav=0.1, freq=0.1, guideStat=1, lin=(0.1, 0.1))
+        elif runId == "67890":
+            return DetectorState(arc=(0.2, 0.2), wav=0.2, freq=0.2, guideStat=1, lin=(0.2, 0.2))
+        return None
+
+    localDataService.readDetectorState = mock.Mock(side_effect=mock_readDetectorState)
+
+    stateSHA1 = UNCHANGING_STATE_ID
```

Comment:
You could also do
``` python
localDataService.readDetectorState = mock.Mock(side_effect = [
        DetectorState(arc=(0.1, 0.1), wav=0.1, freq=0.1, guideStat=1, lin=(0.1, 0.1)),
        DetectorState(arc=(0.2, 0.2), wav=0.2, freq=0.2, guideStat=1, lin=(0.2, 0.2)),
    ]
)
```
It might be a bit more readable.